components/brave_origin/brave_origin_policy_manager.h

```diff
@@ -0,0 +1,88 @@
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ORIGIN_BRAVE_ORIGIN_POLICY_MANAGER_H_
+#define BRAVE_COMPONENTS_BRAVE_ORIGIN_BRAVE_ORIGIN_POLICY_MANAGER_H_
+
+#include <optional>
+
+#include "base/containers/flat_map.h"
+#include "base/memory/raw_ptr.h"
+#include "base/observer_list.h"
+#include "base/observer_list_types.h"
+#include "brave/components/brave_origin/brave_origin_policy_info.h"
+
+class PrefService;
+
+namespace base {
+template <typename T>
+class NoDestructor;
+}
+namespace brave_origin {
+
+// Singleton that holds BraveOrigin preference definitions and manages
+// access to policy values from local state. This completely abstracts away
+// the local state management from policy providers.
+class BraveOriginPolicyManager {
+ public:
+  // Observer interface for objects that need to be notified when
+  // BraveOrigin policies are loaded/changed.
+  class Observer : public base::CheckedObserver {
+   public:
+    // Called when BraveOrigin policies become available or are updated.
+    virtual void OnBraveOriginPoliciesReady() = 0;
+  };
+
+  static BraveOriginPolicyManager* GetInstance();
+
+  // Initialize with pref definitions mappings from browser layer and local
+  // state
+  void Init(BraveOriginPolicyMap&& browser_policy_definitions,
+            BraveOriginPolicyMap&& profile_policy_definitions,
+            PrefService* local_state);
+
+  // Add/remove observers for policy readiness notifications
+  void AddObserver(Observer* observer);
+  void RemoveObserver(Observer* observer);
+
+  // Get policy value for a specific pref, using local state value or default
+  // For profile-scoped prefs, profile_id is required to construct the correct
+  // key. Returns std::nullopt if the pref is unknown.
+  std::optional<bool> GetPolicyValue(const std::string& pref_name,
+                                     const std::string& profile_id = "") const;
+
+  // Get browser-level policy definitions (metadata only)
+  const BraveOriginPolicyMap& GetBrowserPolicyDefinitions() const;
+
+  // Get profile-level policy definitions for a specific profile (metadata only)
+  const BraveOriginPolicyMap& GetProfilePolicyDefinitions(
+      std::string_view profile_id) const;
+
+  // Check if the singleton has been initialized
+  bool IsInitialized() const;
+
+  // Helper function to get pref info from pref definitions
+  const BraveOriginPolicyInfo* GetPrefInfo(const std::string& pref_name);
+
+  BraveOriginPolicyManager(const BraveOriginPolicyManager&) = delete;
+  BraveOriginPolicyManager& operator=(const BraveOriginPolicyManager&) = delete;
+
+ private:
+  friend base::NoDestructor<BraveOriginPolicyManager>;
+  friend class BraveOriginPolicyManagerTest;
+
+  BraveOriginPolicyManager();
+  ~BraveOriginPolicyManager();
+
+  bool initialized_ = false;
+  BraveOriginPolicyMap browser_policy_definitions_;
+  BraveOriginPolicyMap profile_policy_definitions_;
+  raw_ptr<PrefService> local_state_ = nullptr;
```

Comment:
During browser shutdown, are there any issues with holding a `PrefService` pointer in a singleton? I don't think there are, but thought I'd check.